include/logging/log_core.h

```diff
@@ -301,40 +301,40 @@ static inline char z_log_minimal_level_to_char(int level)
 /******************************************************************************/
 /****************** Macros for hexdump logging ********************************/
 /******************************************************************************/
-#define __LOG_HEXDUMP(_level, _id, _filter, _data, _length, _str)	       \
-	do {								       \
-		bool is_user_context = _is_user_context();		       \
-									       \
-		if (Z_LOG_CONST_LEVEL_CHECK(_level)) {			       \
-			if (IS_ENABLED(CONFIG_LOG_MINIMAL)) {		       \
-				Z_LOG_TO_PRINTK(_level, "%s", _str);	       \
-				log_minimal_hexdump_print(_level, _data,       \
-							  _length);	       \
-			} else if (is_user_context ||			       \
-				   (_level <= LOG_RUNTIME_FILTER(_filter))) {  \
-				struct log_msg_ids src_level = {	       \
-					.level = _level,		       \
-					.domain_id = CONFIG_LOG_DOMAIN_ID,     \
-					.source_id = _id,		       \
-				};					       \
-									       \
-				if (is_user_context) {			       \
-					log_hexdump_from_user(src_level, _str, \
-							      _data, _length); \
-				} else if (IS_ENABLED(CONFIG_LOG_IMMEDIATE)) { \
-					log_hexdump_sync(src_level, _str,      \
-							 _data, _length);      \
-				} else {				       \
-					log_hexdump(_str, _data, _length,      \
-						    src_level);		       \
-				}					       \
-			}						       \
-		}							       \
+#define __LOG_HEXDUMP(_level, _id, _filter, _data, _length, _str)				\
+	do {											\
+		bool is_user_context = _is_user_context();					\
+												\
+		if (Z_LOG_CONST_LEVEL_CHECK(_level)) {						\
+			if (IS_ENABLED(CONFIG_LOG_MINIMAL)) {					\
+				Z_LOG_TO_PRINTK(_level, "%s", _str);				\
+				log_minimal_hexdump_print(_level, (const char *)_data,		\
+							  _length);				\
```

Comment:
It's bad to have to turn on "ignore whitespace" to see what actually changed in this macro.  This can become:
```
				log_minimal_hexdump_print(_level,   \
				                          (const char *)_data, \
							  _length); \
```
i.e. leave the backslash at the end of the lines at its original position for the lines that were not changed (so they don't show up in the diff at all), and where necessary break the line that had a cast added so it is as short as possible without splitting the cast expression across lines.  On those lines the backslash can be misaligned with one intervening space.

At least that's how I'd do it.

Reply:
I didn't want to necessarily assume that the changes in what's on each line would be more permissible than just changing the whitespace, thus my decision to do the latter. I'm happy to take the other approach, though.